src/components/product-modal/ProductContainer.tsx

```diff
@@ -0,0 +1,53 @@
+import ProductCarousel from "../product/carousel/ProductCarousel";
+import { Box, Button, SimpleGrid, Stack } from "@chakra-ui/react";
+import ProductSummary from "../product/ProductSummary";
+import ProductExtensions from "../product/ProductExtensions";
+import CartActions from "../product/CartActions";
+import { IBase } from "../../lib/product-types";
+import { ReactElement } from "react";
+import Link from "next/link";
+
+interface IProductContainer {
+  productBase: IBase;
+  handleAddToCart: () => void;
+  children?: ReactElement;
+}
+
+export default function ProductContainer({
+  productBase: { product, main_image, otherImages },
+  handleAddToCart,
+  children,
+}: IProductContainer): JSX.Element {
+  const { extensions } = product.attributes;
+  return (
+    <SimpleGrid
+      columns={{ base: 1, lg: 2 }}
+      spacing={{ base: 8, md: 10 }}
+      py={{ base: 8 }}
+    >
+      {main_image && (
+        <ProductCarousel images={otherImages} mainImage={main_image} />
+      )}
+      <Stack spacing={{ base: 6, md: 10 }}>
+        <ProductSummary product={product} />
+        {extensions && <ProductExtensions extensions={extensions} />}
+        {children}
+        <Box>
+          <CartActions handleAddToCart={handleAddToCart} />
+          <Link href={`/products/${product.id}`} passHref>
+            <Button
+              _hover={{
+                color: "brand.primary",
+              }}
+              m="10px auto"
```

Comment:
very minor point but it would be good if we can avoid using px where we can

Reply:
@samblacklock my bad, thanks